progress/progress.go

```diff
@@ -12,16 +12,29 @@ import (
 	"github.com/moby/buildkit/util/progress/progressui"
 )
 
+// Progress defines functions for a buildkit progress display.
 type Progress interface {
-	Release()
-	Sync()
-	Pause()
-	Resume()
+	// Channel returns a new buildkit client.SolveStatus channel used for
+	// buildkit to send messages to the progress display.
 	Channel(opts ...ChannelOption) chan *client.SolveStatus
+	// Label returns a new Progress where all messages will have the provided
+	// label prepended to displayed messages.
 	Label(string) Progress
+	// Pause will sync the Progress, then prevent further messages from being
+	// written to the display channels.  Resume MUST be called after Pause.
+	Pause()
+	// Resume will unblock the display channels so messages will start
+	// displaying after a Pause.
+	Resume()
+	// Release MUST be called for each new Progress. This releases resources
```

Comment:
Is this basically a `Close` method? Saying it needs to be called for each _new_ `Progress` seems a little confusing.